File: backend/v1/src/services/all_subs.py
import connect_db
from flask import Flask, jsonify


def all_subs_data():
    connection = connect_db.connect()
    cur = connection.cursor()

    query = (
        "SELECT pubsub_name FROM {table} WHERE pubsub_name is not NULL ORDER BY on_sale"
    )
    cur.execute(query.format(table=connect_db.get_table()))

    records = cur.fetchall()

    data = {}

    """
    Creates a primary catagory
    """
    data["All_subs".lower()] = []

    """
    Create a default JSON structure
    """
    for sub in records:
        data["All_subs".lower()].append({"name": sub[0]})
    response = jsonify(data["All_subs".lower()])
    response.headers.add("Access-Control-Allow-Origin", "*")
    return response

File: backend/v2/src/services/allSubs.go
package services

import (
	"fmt"

	"github.com/gin-gonic/gin"
)

type Subs struct {
	All_Subs []Sub
}
type Sub struct {
	name string
}

// Grabs all the subs currently available from the database
// And packages it into a json file
func AllSubs(c *gin.Context) {
	connection := ConnectToDB()
	fmt.Println(connection)
	// query := ("SELECT pubsub_name FROM %s" +
	// 	" WHERE pubsub_name is not NULL ORDER BY on_sale")
	// queryFormat := fmt.Sprintf(query, GetTable())
	// cur, _ := connection.pi
	// fmt.Println(cur)
	// c.JSON(200, gin.H{
	// 	"message": "pong",
	// })
}
